apps/teams-test-app/src/components/PagesBackStackAPIs.tsx

```diff
@@ -61,16 +62,26 @@ const PagesBackStackAPIs = (): ReactElement => {
     setAddStatesValue('total States: ' + newNumStates);
   }, [totalStates, setTotalStates, setAddStatesValue]);
 
+  const registerBackButtonHandlerHelper = (): boolean => {
+    if (totalStates > 0) {
+      const newNumStates = totalStates - 1;
+      setTotalStates(newNumStates);
+      setRegisterBackButtonHandlerValue('back button clicked. total remaining state: ' + newNumStates);
+      return true;
+    }
+    return false;
+  };
+
   const onRegisterBackButtonHandler = React.useCallback(() => {
-    pages.backStack.registerBackButtonHandler((): boolean => {
-      if (totalStates > 0) {
-        const newNumStates = totalStates - 1;
-        setTotalStates(newNumStates);
-        setRegisterBackButtonHandlerValue('back button clicked. total remaining state: ' + newNumStates);
-        return true;
-      }
-      return false;
-    });
+    if (getTestBackCompat()) {
```

Comment:
This breaks our current way of doing things, but I'm not sure how to go around it.

Reply:
@MarcinJuraszek Can you please take a look here?

Reply:
I think this is fine. It's a non-standard API which is why it's not using the `ApiWithTextInput` component. I don't think we should spend any more time than necessary to try to get it look/feel 'better'. This seems totally appropriate to me.